### PR TITLE
Simplify strings in conventions for pure functions

### PR DESCRIPTION
**Pull request by tovrstra**
The character for the angular momentum is omitted because it is redundant. With this change, the conventions are no longer limited in angular momentum by the availability of alphabetical characters.

Some mistakes in the docstring of `MolecularBasis` are also fixed. More elaborate documentation will be added in a separate PR. See #131

### CODE REVIEW

File: iodata/basis.py [hunk 1]
```diff
@@ -150,17 +150,29 @@ class MolecularBasis(NamedTuple):
...
-                # or Wikipedia-ordered real solid spherical harmonics
-                # c = cosine-like
-                # s = sine-like
-                (2, 'p'): ['dc2', 'dc1', 'dc0', '-ds1', '-ds2'],
-                ...
+                ### Conventions for pure functions.
+                # The notation is referring to real solid spherical harmonics.
+                # See https://en.wikipedia.org/wiki/Solid_harmonics#Real_form
+                # c = cosine of the azimuthal angle
+                # s = sine of the azimuthal angle
```
Comment (kimt33) [inline iodata/basis.py]: Would it be alright to add `number = magnetic quantum number` (or the correct term) here? It would be helpful to have a definition of each component of the notation.

File: iodata/basis.py [hunk 2]
```diff
@@ -150,17 +150,29 @@ class MolecularBasis(NamedTuple):
...
+                # For example, wikipedia-ordered real spherical harmonics,
+                # see https://en.wikipedia.org/wiki/Spherical_harmonics#Real_form
+                (2, 'p'): ['s2', 's1', 'c0', 'c1', 'c2'],
+                # Different quantum-chemistry codes may use incompatible
+                # orderings and sign conventions. E.g. Molden files written
+                # by Orca use the following convention for pure f functions:
+                (3, 'p'): ['c0', 'c1', 's1', 'c2', 's2', '-c3', '-s3'],
+                # Note the sign change of the last two basis functions.
```
Comment (kimt33) [inline iodata/basis.py]: Could this elaborate more on the sign change? Something like `Note that the minus sign in the last two basis functions denotes that the signs of these harmonics have been changed.`

I thought it'd be nice to have a more explicit definition, but maybe it's a bit wordy and is redundant. So whatever you think is best.

File: iodata/basis.py (hunk 1 above)
Comment (tovrstra) [reply iodata/basis.py]: good point.

File: iodata/basis.py (hunk 2 above)
Comment (tovrstra) [reply iodata/basis.py]: I'll add it. I'm going to write out all the equations and details in documentation in the next PR, but it does not hurt to make it a bit more explicit here.
